collections/_posts/2022-11-28-CurriculumLearningByDIH.md

```diff
@@ -0,0 +1,110 @@
+
+## Curriculum learning
+
+As the DIH represents the hardness of samples it seems natural to base some curriculum learning on it (DIHCL). They keep training the model on samples whith large DIH that have historically been hard since the model does not perform well on them and revisit easy samples less frequently because the model is more likely to stay at those samples’ minima. 
+At each training step a subset of samples is selected according to their DIH values, given the probability $$p_{t,i} \propto h(r_{t-1}(i))$$ where $$h(·)$$ is a monotone non-decreasing function. 
+* DIHCL-Rand: data is sample proportionnaly to DIH $$h(r_t(i)) = r_t(i)$$
+* DIHCL-Exp: trade-off exploration/exploitation based on softmax value $$h(r_t(i)) = exp [\sqrt{2*log(n/n)}*r_t(i)], a_t(i) \leftarrow a_t(i)/p_{t,i} \forall i \in S_t$$
+* DIHCL-Beta: Beta prior distribution to balance exploration and exploitation $$h(r_t(i)) \leftarrow Beta(r_t(i),c-r_t(i))$$ with $$c > r_t(i)$$
+
+The DIH of selected samples is updated with their instantaneous hardness normalized by the learning rate as it varies during training. During the first few epochs the whole dataset is used to set a correct DIH to each samples and then starts to select training samples, the subset size gradually deacrease during training (empirical study made to have optimal parameters).
+
+Comparison is made with random baseline, SPL (based on instantaneous hardness) and MCL (Minmax curriculum learning[^3]).
+
+## Results
+
+
+![](/collections/images/CL_DIH/accuracy_table.jpg)
+
+Better accuracy with DIHCL.
+
+![](/collections/images/CL_DIH/accuracy_curves.jpg)
+
+DIHCL is more stable and reach its best performance sooner than SPL and MCL.
+
+![](/collections/images/CL_DIH/training_time.jpg)
+
+Training time reduced with DIHCL.
+
+## Ablation studies
+
+![](/collections/images/CL_DIH/ablation_study.jpg)
+
+More experiments are presented in the appendix of the article to explain the decisions on the parameters. The more important settings concerns the discount factor $$\gamma$$ to calculate the DIH, the number of warm starting epochs and the cosine for the cyclic learning rate.
+
+# Conclusions
+
+The article introduces a new metric to implement curriculum learning that is doubly effective as it reduces the training time and does not weed a previous training to be set. It wound be intersting to compare the performances to other types of CL and not only SPL and MCL like teacher-student CL that is another great CL technique.
+
+# References
+
+[^1]: Y. Bengio et al. Curriculum learning. Journal of the American Podiatry Association 60(1):6. January 2009. DOI:10.1145/1553374.1553380
+[^2]: M. Kumar et al. Self-Paced Learning for Latent Variable Models. Procedings of the 23rd conference on Neural Information Processing Systems. 2010. DOI:10.5555/2997189.2997322
+[^3]: T. Zhou and J. Bilmes. Minimax curriculum learning: Machine teaching with desirable difficulties and scheduled diversity. In ICLR. February 2018.
```

Comment:
It would be great if you could format these references as hyperlinks directly to the papers :wink: